features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/internal/DatadogFlagsClient.kt

```diff
@@ -400,14 +404,66 @@ internal class DatadogFlagsClient(
     }
 
     private fun <T : Any> trackResolution(resolution: InternalResolution.Success<T>) {
-        if (resolution.flag.doLog) {
+        trackResolution(resolution.flagKey, resolution.flag, resolution.value, resolution.context)
+    }
+
+    private fun <T : Any> trackResolution(
+        flagKey: String,
+        flag: UnparsedFlag,
+        flagValue: T,
+        context: EvaluationContext
+    ) {
+        if (flag.doLog) {
             if (flagsConfiguration.trackExposures) {
-                writeExposureEvent(resolution.flagKey, resolution.flag, resolution.context)
+                writeExposureEvent(flagKey, flag, context)
             }
             if (flagsConfiguration.rumIntegrationEnabled) {
-                logEvaluation(resolution.flagKey, resolution.value)
+                logEvaluation(flagKey, flagValue)
+            }
+        }
+    }
+
+    // endregion
+
+    // region Internal APIs exposed through _FlagsInternalProxy
+
+    /**
+     * Retrieves a snapshot of all flag assignments.
+     *
+     * Supposed to be used by internal Datadog packages to get flags state snapshot for a given evaluation context.
+     *
+     * @return A map of flag key to precomputed flag, or null if no flags are available.
+     */
+    internal fun getFlagAssignmentsSnapshot(): Map<String, UnparsedFlag>? = flagsRepository.getFlagsSnapshot()
+
+    /**
+     * Tracks the evaluation of a flag from an exact flags state snapshot.
+     *
+     * Supposed to be used by internal Datadog packages to track flag evaluations from an exact flags state snapshot.
+     */
+    internal fun trackFlagSnapshotEvaluation(flagKey: String, flag: UnparsedFlag, context: EvaluationContext) {
+        val flagValue = parseFlagValueString(flag)
+
+        trackResolution(flagKey, flag, flagValue, context)
+    }
+
+    private fun parseFlagValueString(flag: UnparsedFlag): Any {
+        val value: Any = when (flag.variationType) {
+            VariationType.BOOLEAN.value -> flag.variationValue.lowercase(Locale.US).toBooleanStrictOrNull()
+                ?: flag.variationValue
+            VariationType.STRING.value -> flag.variationValue
+            VariationType.INTEGER.value -> flag.variationValue.toIntOrNull() ?: flag.variationValue
+            VariationType.NUMBER.value, VariationType.FLOAT.value -> flag.variationValue.toDoubleOrNull()
+                ?: flag.variationValue
+            VariationType.OBJECT.value -> try {
+                JSONObject(flag.variationValue)
+            } catch (_: JSONException) {
+                flag.variationValue
```

Comment:
would it be useful to make `featureSdkCore.internalLogger.log` call here to log exception instead of swallowing it?